### DISCUSSION TITLE
Clicking song in Leaderboards redirect to SongsLeaderboard

### DISCUSSION THREAD

**Discussion by alexissyp** [Australian-DDR-Events/aus-ddr-poc#79] [Q&A]
Hi, @nhivn @Chris-SG @Chill-X . I'm trying to figure out how this works. 

https://github.com/Australian-DDR-Events/aus-ddr-events-frontend/blob/8610a1ff28acd2f39a1058bf46841cc442952113/src/components/router/index.tsx#L75-L82
> :id is a placeholder.

https://github.com/Australian-DDR-Events/aus-ddr-events-frontend/blob/8610a1ff28acd2f39a1058bf46841cc442952113/src/components/router/index.tsx#L79
> Share state(id) with Leaderboard component.

https://github.com/Australian-DDR-Events/aus-ddr-events-frontend/blob/8610a1ff28acd2f39a1058bf46841cc442952113/src/features/leaderboard/index.tsx#L91-L93
> This replace the placeholder with song's id?

https://github.com/Australian-DDR-Events/aus-ddr-events-frontend/blob/8610a1ff28acd2f39a1058bf46841cc442952113/src/features/leaderboard/index.tsx#L28
> This makes it so that when songId is triggered by clicking Box https://github.com/Australian-DDR-Events/aus-ddr-events-frontend/blob/8610a1ff28acd2f39a1058bf46841cc442952113/src/features/leaderboard/index.tsx#L91-L93, it'll show SongLeaderboard? 

May I know if what I think is correct or is there something else that plays a part in this? Thanks for reading.

**Comment by Chris-SG** [comment]
So the `leaderboard` component actually has two separate renders. One is with a `songId` parameter, one without. This is the difference between the listing of all the songs, vs all the scores for a particular song.

The `setLocation(...)` calls are using whats known as a hook (in this case, part of [wouter](https://github.com/molefrog/wouter)). This just changes the url in your browser to something else. This is how we perform navigation between pages in most cases (it prevents reloading the page and instead just changes the url, and our router component handles the changes).

The `:id` bit in the router isn't exactly a placeholder. They are part of wouter as well, and are referred to as dynamic sections. It's pretty much a fancy name for a wildcard, but the value of it can be referenced in the `params` object (eg. `params.id`).

Ping me on discord if any of this is confusing.

**Reply by alexissyp** [reply]
Thanks for informative explanation 👍 !
